### PR TITLE
fix(apt): use proper apt config locations

### PR DESCRIPTION
**Pull request by Elsie19**
## Checklist

- [x] I confirm that I have read the [contributing guidelines](https://github.com/pacstall/pacstall/blob/develop/CONTRIBUTING.md), and this pull request is abiding by all the clauses stated in the guideline.

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -75,7 +75,9 @@ fi
 echo -e "${PACYELLOW}┌────────────────────────┐\n│   ${PACCYAN}Pacstall Installer${PACYELLOW}   │\n└────────────────────────┘${NC}"
 
 echo
-if [[ -z "$(find -H /var/lib/apt/lists -maxdepth 0 -mtime -7)" ]]; then
+eval "$(apt-config shell Cache Dir::Cache)"
+eval "$(apt-config shell List Dir::State::Lists)"
+if [[ -z "$(find -H "${Cache}/${List}" -maxdepth 0 -mtime -7)" ]]; then
```
Comment (D-Brox) [inline install.sh:80]:
```suggestion
if [[ -z "$(find -H "/${Cache}/${List}" -maxdepth 0 -mtime -7)" ]]; then
```

File: pacstall [hunk 2]
```diff
@@ -628,7 +628,10 @@ function getMasks_offending_pkg() {
 }
 
 # run sudo apt update if it's been more than a week
-[[ -z "$(find -H /var/lib/apt/lists -maxdepth 0 -mtime -7)" ]] && sudo apt-get update -qq --allow-releaseinfo-change
+eval "$(apt-config shell Cache Dir::Cache)"
+eval "$(apt-config shell List Dir::State::Lists)"
+[[ -z "$(find -H "${Cache}/${List}" -maxdepth 0 -mtime -7)" ]] && sudo apt-get update -qq --allow-releaseinfo-change
```
Comment (D-Brox) [inline pacstall:633]:
```suggestion
[[ -z "$(find -H "/${Cache}/${List}" -maxdepth 0 -mtime -7)" ]] && sudo apt-get update -qq --allow-releaseinfo-change
```

**Review by D-Brox — CHANGES_REQUESTED** [review]

File: install.sh (hunk 1 above)
Comment (oklopfer) [inline install.sh:80]: I dont like the evals
```suggestion
Cache="$(apt-config shell Cache Dir::Cache | cut -d"'" -f2)"
List="$(apt-config shell List Dir::State::Lists | cut -d"'" -f2)"
if [[ -z "$(find -H "/${Cache}/${List}" -maxdepth 0 -mtime -7)" ]]; then
```

File: pacstall (hunk 2 above)
Comment (oklopfer) [inline pacstall:633]:
```suggestion
Cache="$(apt-config shell Cache Dir::Cache | cut -d"'" -f2)"
List="$(apt-config shell List Dir::State::Lists | cut -d"'" -f2)"
[[ -z "$(find -H "/${Cache}/${List}" -maxdepth 0 -mtime -7)" ]] && sudo apt-get update -qq --allow-releaseinfo-change
```

**Comment by Elsie19** [comment]
The evals are safe, `apt-config shell` is literally made for evaling.

-------- Original Message --------

**Comment by D-Brox** [comment]
@Elsie19 it still needs the "/" tho, since it evals to `var/apt/lists` on normal circumstances

File: pacstall (hunk 2 above)
Comment (Elsie19) [reply pacstall:633]: No.

**Review by oklopfer — APPROVED** [review]

**Comment by oklopfer** [comment]
turns out this was a regression

**Pull request merged**